app/src/main/java/vn/sun/music_68/ui/home/adapter/GenreAdapter.java

```diff
@@ -0,0 +1,91 @@
+package vn.sun.music_68.ui.home.adapter;
+
+import android.content.Context;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.ImageView;
+import android.widget.TextView;
+
+import androidx.annotation.NonNull;
+import androidx.recyclerview.widget.RecyclerView;
+
+import com.bumptech.glide.Glide;
+
+import vn.sun.music_68.R;
+import vn.sun.music_68.base.BaseRecyclerViewAdapter;
+import vn.sun.music_68.data.model.Track;
+
+public class GenreAdapter extends BaseRecyclerViewAdapter<Track, GenreAdapter.ViewHolder> {
+    private GenreClickListener mListener;
+    private LayoutInflater mLayoutInflater;
+    private static final String[] GENRES_NAME = {"ALL MUSIC", "ALL AUDIO", "ALTERNATIVEROCK", "AMBIENT",
+            "CLASSICAL", "COUNTRY"};
+    private static final int[] GENRES_ARTWORK = {
+            R.drawable.genres_all_music,
+            R.drawable.genres_all_audio,
+            R.drawable.genres_alternativerock,
+            R.drawable.genres_ambient,
+            R.drawable.genres_classical,
+            R.drawable.genres_country
+    };
+
+    public GenreAdapter(Context context, GenreClickListener listener) {
+        mLayoutInflater = LayoutInflater.from(context);
+        mListener = listener;
+    }
+
+    @NonNull
+    @Override
+    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
+        View view = LayoutInflater.from(parent.getContext())
+                .inflate(R.layout.item_genres, parent, false);
+        return new ViewHolder(view, mListener);
+    }
+
+    @Override
+    public void onBindViewHolder(@NonNull ViewHolder viewHolder, int position) {
+        viewHolder.bindData(position);
+    }
+
+    @Override
+    public int getItemCount() {
+        return GENRES_NAME.length;
+    }
+
+    public static class ViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener {
+        private ImageView mGenreImage;
+        private TextView mGenreName;
+        private GenreClickListener mListener;
+
+        public ViewHolder(@NonNull View itemView, GenreClickListener listener) {
+            super(itemView);
+            mListener = listener;
+            initView();
+        }
+
+        public void initView() {
+            mGenreImage = itemView.findViewById(R.id.image_artwork);
+            mGenreName = itemView.findViewById(R.id.text_artist);
+        }
+
+        @Override
+        public void onClick(View v) {
+            mListener.onItemClickListener(this.getLayoutPosition());
+        }
+
+        public void bindData(int position) {
+            Glide.with(itemView.getContext())
+                    .load(GENRES_ARTWORK[position])
+                    .centerCrop()
+                    .placeholder(R.drawable.genres_all_audio)
+                    .into(mGenreImage);
+            mGenreName.setText(GENRES_NAME[position]);
+            itemView.setOnClickListener(this);
```

Comment:
```suggestion
             public ViewHolder(@NonNull View itemView, GenreClickListener listener) {
	            super(itemView);
	            mListener = listener;
	            itemView.setOnClickListener(this);
	            initView();
	        }
```
vì ở bind data sẽ phải set đi set lại nhiều lần